src/Tempest/EventBus/src/CallableEventHandler.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\EventBus;
+
+use Closure;
+use Tempest\Container\Container;
+
+final class CallableEventHandler
+{
+    public function __construct(
+        public null|string|object $event,
+        public EventHandler|Closure $handler,
+    ) {
+    }
+
+    public function normalizeCallable(Container $container): Closure
+    {
+        if ($this->handler instanceof Closure) {
+            return function (string|object $event) use ($container): void {
+                $container->invoke($this->handler, event: $event);
+            };
+        }
+
+        return function (string|object $event) use ($container): void {
+            $this->handler->handler->invokeArgs(
+                $container->get($this->handler->handler->getDeclaringClass()->getName()),
```

Comment:
Do I get it right that from now on we can also inject any dependency into event handler methods because of this? That was actually on my TODO somewhere, but I never got to it :p

Reply:
Actually, not right now—only on `EventBus#listen`... but this would be fairly easy to implement, I just forgot about this. The change is as simple as this:

```diff
return function (string|object $event) use ($container): void {
+     $container->invoke($this->handler->handler, event: $event);
-     $this->handler->handler->invokeArgs(
-         $container->get($this->handler->handler->getDeclaringClass()->getName()),
-        [$event],
-     );
```